### PR TITLE
add an empty create() method for UICCTextField

### PR DESCRIPTION
**Pull request by drelaptop**
A create method with parameters is used In an empty init method, It will do something useless, and cause issues like https://github.com/cocos2d/cocos2d-x/issues/18403

### CODE REVIEW

File: cocos/ui/UITextField.cpp [hunk 1]
```diff
@@ -46,6 +46,20 @@ UICCTextField::~UICCTextField()
 {
 }
 
+UICCTextField * UICCTextField::create()
+{
+    UICCTextField *pRet = new (std::nothrow) UICCTextField();
```
Comment (minggo) [inline cocos/ui/UITextField.cpp]: pRet -> ret
we don't use Hungarian Notation now. Old codes need to be modified when meet it.

File: cocos/ui/UITextField.h [hunk 2]
```diff
@@ -58,6 +58,13 @@ class CC_GUI_DLL UICCTextField: public TextFieldTTF, public TextFieldDelegate
     
     virtual void onEnter() override;
     
+    /**
+     *@brief Create an empty UICCTextField.
+     
+     *@return A UICCTextField instance.
+     */
+    static UICCTextField* create();
```
Comment (minggo) [inline cocos/ui/UITextField.h]: static function should be declared before all other functions. Please refer to [coding style][https://github.com/cocos2d/cocos2d-x/blob/v3/docs/CODING_STYLE.md].

Comment (drelaptop) [reply cocos/ui/UITextField.h]: OK, I will look at code style, and change it

File: cocos/ui/UITextField.cpp [hunk 3]
```diff
@@ -32,6 +32,20 @@ NS_CC_BEGIN
...
+UICCTextField * UICCTextField::create()
+{
+    UICCTextField *ret = new (std::nothrow) UICCTextField();
+
+    if(ret)
+    {
+        ret->autorelease();
+        return ret;
+    }
+    CC_SAFE_DELETE(ret);
```
Comment (minggo) [inline cocos/ui/UITextField.cpp]: Is it necessary? ret is nullptr here, may change the code to 
```c++
if (ret)
    ret->autorelease();

return ret;
```

**Pull request merged**